internal/controllers/topology/cluster/structuredmerge/serversidepathhelper_test.go

```diff
@@ -655,15 +655,25 @@ func TestServerSideApplyWithDefaulting(t *testing.T) {
 			}()
 
 			// Run defaulting on the KubeadmConfigTemplate (triggered by an "external controller")
+			// Note: We have to retry this with eventually as it seems to take a bit of time until
+			// the webhook is active.
 			if tt.defaultOriginal {
-				patchKCT := kct.DeepCopy()
-				if patchKCT.Labels == nil {
-					patchKCT.Labels = map[string]string{}
-				}
-				patchKCT.Labels["trigger"] = "update"
-				g.Expect(env.Patch(ctx, patchKCT, client.MergeFrom(kct))).To(Succeed())
-				// Ensure patchKCT was defaulted.
-				g.Expect(patchKCT.Spec.Template.Spec.Users).To(Equal([]bootstrapv1.User{{Name: "default-user"}}))
+				g.Eventually(ctx, func(g Gomega) {
+					patchKCT := &bootstrapv1.KubeadmConfigTemplate{}
+					g.Expect(env.Get(ctx, client.ObjectKeyFromObject(kct), patchKCT))
+
+					if patchKCT.Labels == nil {
+						patchKCT.Labels = map[string]string{}
+					}
+					patchKCT.Labels["trigger"] = "update"
+
+					g.Expect(env.Patch(ctx, patchKCT, client.MergeFrom(kct))).To(Succeed())
+
+					// Ensure patchKCT was defaulted.
+					g.Expect(env.Get(ctx, client.ObjectKeyFromObject(kct), patchKCT))
+					g.Expect(patchKCT.Spec.Template.Spec.Users).To(Equal([]bootstrapv1.User{{Name: "default-user"}}))
+				}, 5*time.Second).Should(Succeed())
```

Comment:
Q on this, but not really for this PR. Wasn't there an issue recently with incorrectly specified timeouts / polling frequencies with Gomega's eventually?

I can't remember exactly what the issue was though.

Reply:
Nice catch, thank you.

But turns out my previous take was wrong.

gomega is doing the following
```go
	if len(intervals) > 0 {
		timeoutInterval, err = toDuration(intervals[0])
		if err != nil {
			g.Fail(err.Error(), offset+baseOffset)
		}
	}
	if len(intervals) > 1 {
		pollingInterval, err = toDuration(intervals[1])
		if err != nil {
			g.Fail(err.Error(), offset+baseOffset)
		}
	}
```

Which means if you only set one interval you only set the timeoutInterval and not the pollingInterval.

I thought this means it doesn't actually retry. I justed tested it and it turns out it does retry. Going through some gomega internals reveals that gomega is using a default poll interval of 100ms.

So we should be good